### PR TITLE
feat: sign docker images with cosign

### PR DESCRIPTION
**Pull request by caarlos0**
was a bit sleep deprived and also couldn't sleep, decided to hack this thing that was in the back of my mind for a while now.

also improved the signing pipe docs a bit.

### CODE REVIEW

**Review by radeksimko — COMMENTED** [review]
I admit I haven't done much container image packaging myself lately, but it looks like a useful feature!

One concern I would have is that most of the examples read key material from the filesystem, which doesn't seem like a good practice for a CI system usually? It appears that cosign has some mechanism for encrypting the private key but it's unclear to me at which point would you decrypt it and generally what risks are associated with exposure of any of these?

Naturally not everyone will opt into storing it in a KMS or Vault as that comes with its own level of complexity but I do think that encouraging use of environment variables and/or something like GitHub Actions secrets would be nice, if cosign supports that.

File: www/docs/customization/sign.md [hunk 1]
```diff
@@ -30,30 +30,34 @@ To customize the signing pipeline you can use the following options:
...
-    # name/template of the signature file.
-    # '${artifact}' is the path to the artifact that should be signed.
+    # Name/template of the signature file.
+    #
+    # Available environment variables:
+    # - '${artifact}': the path to the artifact that will be signed
+    # - '${artifactID}': the ID of the artifact that will be signed
     #
-    # defaults to `${artifact}.sig`
+    # Defaults to empty.
```
[the remark this answers is not in the document]
Comment (caarlos0) [reply www/docs/customization/sign.md:43]: oops, bad docs only, default is still the same.

**Comment by caarlos0** [comment]
> Naturally not everyone will opt into storing it in a KMS or Vault as that comes with its own level of complexity but I do think that encouraging use of environment variables and/or something like GitHub Actions secrets would be nice, if cosign supports that.

according to [this](https://blog.sigstore.dev/cosign-signed-container-images-c1016862618?gi=c9528e13df9e), it is safe to commit the private key... maybe @dlorenc can correct me if I understood that wrong.

**Comment by dlorenc** [comment]
It's safe to commit it with a reasonably strong password. That said, both can be stored in GitHub secrets if you prefer.

**Pull request merged**